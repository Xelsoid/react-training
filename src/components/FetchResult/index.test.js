import React from 'react';
import ComponentExample from './index';

describe('FetchResult test', () => {
  it('renders correctly', () => {
    const component = shallow(<ComponentExample />);
    expect(component).toMatchSnapshot();
  });
});
