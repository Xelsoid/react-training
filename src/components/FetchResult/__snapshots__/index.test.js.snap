// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchResult test component doesnt render 1`] = `
<FetchResult
  error={Object {}}
  handlerId=""
  loading={Object {}}
/>
`;

exports[`FetchResult test renders correctly Child component 1`] = `
<FetchResult
  error={Object {}}
  handlerId=""
  loading={Object {}}
>
  <ChildExample>
    <h1>
       Child test component 
    </h1>
  </ChildExample>
</FetchResult>
`;

exports[`FetchResult test renders correctly Error 1`] = `
<FetchResult
  error={
    Object {
      "MOVIE": true,
    }
  }
  handlerId="MOVIE"
  loading={Object {}}
>
  <FetchError>
    <div
      className="fetch-error"
    >
      Request error happened..., please try again.
    </div>
  </FetchError>
</FetchResult>
`;

exports[`FetchResult test renders correctly Loading 1`] = `
<FetchResult
  error={Object {}}
  handlerId="MOVIE"
  loading={
    Object {
      "MOVIE": true,
    }
  }
>
  <Loading>
    <div
      className="loading"
    >
      Loading films... Please wait.
    </div>
  </Loading>
</FetchResult>
`;
